segments/base_segment.go

```diff
@@ -0,0 +1,700 @@
+// Copyright 2020 The Moov Authors
+// Use of this source code is governed by an Apache License
+// license that can be found in the LICENSE file.
+
+package segments
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"unicode"
+	"unicode/utf8"
+)
+
+type BaseSegment struct {
```

Comment:
Can we add a description to the public structs?

Reply:
ok, we need to add a function into segment interface